File: src/main/java/com/thegame/business/model/BuildingType.java
package com.thegame.business.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "building_types", schema = "data")
public class BuildingType {

    @Id
    @Column(name = "building_type_id")
    private Long buildingTypeId;

    @Column(name = "building_name")
    private String buildingName;

    public Long getBuildingTypeId() {
        return buildingTypeId;
    }

    public void setBuildingTypeId(Long buildingTypeId) {
        this.buildingTypeId = buildingTypeId;
    }

    public String getBuildingName() {
        return buildingName;
    }

    public void setBuildingName(String buildingName) {
        this.buildingName = buildingName;
    }
}

File: src/main/java/com/thegame/business/model/BuildingType.kt
package com.thegame.business.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "building_types", schema = "data")
class BuildingType {

    @Id
    @Column(name = "building_type_id")
    var buildingTypeId: Long? = null

    @Column(name = "building_name")
    var buildingName: String? = null

}